scipy/spatial/tests/test_distance.py

```diff
@@ -108,6 +110,21 @@
 eo = {}
 
 
+# if enabled, equivalent to pytest.raises otherwise is a no-op
```

Comment:
As this appears to be used only twice, I suggest inlining the logic with an `if`, which is less code and equally clear in the intent:

```
if output_contiguous:
   with pytest.raises(....):
      _ = .....
```

Reply:
That's not equivalent. The equivalent would be:
```
if output_contiguous:
    with pytest.raises(...):
         # ...
else:
    # Duplicate of code inside with statement above.
```

Reply:
Indeed -- I missed typed that part. Code duplication in unit tests can sometimes be beneficial (like here), to avoid injecting a test condition into a context manager.